tracing-error/src/backtrace.rs

```diff
@@ -0,0 +1,65 @@
+use super::try_bool;
+use crate::layer::WithContext;
+use std::fmt;
+use tracing::{Metadata, Span};
+
+#[derive(Clone)]
+pub struct SpanBacktrace {
+    span: Span,
+}
+
+// === impl SpanBacktrace ===
+
+impl SpanBacktrace {
+    pub fn capture() -> Self {
+        SpanBacktrace {
+            span: Span::current(),
+        }
+    }
+
+    pub fn with_spans(&self, f: impl FnMut(&'static Metadata<'static>, &str) -> bool) {
+        self.span.with_subscriber(|(id, s)| {
+            if let Some(getcx) = s.downcast_ref::<WithContext>() {
+                getcx.with_context(s, id, f);
+            }
+        });
+    }
+}
+
+impl fmt::Debug for SpanBacktrace {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut err = None;
+        let mut span = 0;
+
+        writeln!(f, "span backtrace:")?;
+        self.with_spans(|metadata, fields| {
+            try_bool!(
+                writeln!(f, "{:>4}: {}::{}", span, metadata.target(), metadata.name()),
+                err
+            );
+            if !fields.is_empty() {
+                try_bool!(writeln!(f, "           with {}", fields), err);
```

Comment:
i have a slight preference for expressing the indentation here using a numeric constant or something, rather than literal whitespace, but i don't really care

Reply:
Same as earlier, I just went to the backtrace src to see how they did it and saw that they were using whitespace literals so I just copied them.

Reply:
mmk, i don't really care either way.